### PR TITLE
[receiver/mysql]: add metrics based on events_statements_summary_by_digest table

### PR DESCRIPTION
**Pull request by sumo-drosiek**
**Description:**

Add metrics based on table_lock_waits_summary_by_table table

[mysql docs](https://dev.mysql.com/doc/mysql-perfschema-excerpt/5.7/en/performance-schema-table-lock-waits-summary-by-table-table.html)

This is going to add equivalents of the following telegraf metrics:

```
mysql_perf_schema_read
mysql_perf_schema_read_high_priority
mysql_perf_schema_read_no_insert
mysql_perf_schema_read_normal
mysql_perf_schema_read_with_shared_locks
mysql_perf_schema_write
mysql_perf_schema_write_allow_write
mysql_perf_schema_write_concurrent_insert
mysql_perf_schema_write_low_priority
mysql_perf_schema_write_normal

```

**Link to tracking Issue:** #14138

**Testing:** Units

**Documentation:** auto-generated

### CODE REVIEW

File: receiver/mysqlreceiver/metadata.yaml [hunk 1]
```diff
@@ -279,3 +287,39 @@ metrics:
...
+    attributes: [schema, table_name, write_lock_type]
+  mysql.table_lock_wait.write.time:
+    enabled: true
+    description: The total table lock wait write events times.
+    unit: ns
+    sum:
+      value_type: int
+      monotonic: false
+      aggregation: cumulative
+    attributes: [schema, table_name, write_lock_type]
```
Comment (djaglowski) [inline receiver/mysqlreceiver/metadata.yaml:325]: `mysql.table` seems appropriate for a namespace.

```suggestion
  mysql.table.lock_wait.read.count:
    enabled: true
    description: The total table lock wait read events.
    unit: "1"
    sum:
      value_type: int
      monotonic: false
      aggregation: cumulative
    attributes: [schema, table_name, read_lock_type]
  mysql.table.lock_wait.read.time:
    enabled: true
    description: The total table lock wait read events times.
    unit: ns
    sum:
      value_type: int
      monotonic: false
      aggregation: cumulative
    attributes: [schema, table_name, read_lock_type]
  mysql.table.lock_wait.write.count:
    enabled: true
    description: The total table lock wait write events.
    unit: "1"
    sum:
      value_type: int
      monotonic: false
      aggregation: cumulative
    attributes: [schema, table_name, write_lock_type]
  mysql.table.lock_wait.write.time:
    enabled: true
    description: The total table lock wait write events times.
    unit: ns
    sum:
      value_type: int
      monotonic: false
      aggregation: cumulative
    attributes: [schema, table_name, write_lock_type]
```

File: receiver/mysqlreceiver/scraper_test.go [hunk 2]
```diff
@@ -185,6 +188,47 @@ func (c *mockClient) getIndexIoWaitsStats() ([]IndexIoWaitsStats, error) {
 	return stats, nil
 }
 
+func (c *mockClient) gettableLockWaitEventStatss() ([]tableLockWaitEventStats, error) {
```
Comment (djaglowski) [inline receiver/mysqlreceiver/scraper_test.go:191]:
```suggestion
func (c *mockClient) getTableLockWaitEventStatss() ([]tableLockWaitEventStats, error) {
```

**Review by djaglowski — COMMENTED** [review]
These metrics could potentially be very high cardinality (e.g. when there are many schemas and/or tables). Can we disable them by default?

**Comment by portertech** [comment]
@djaglowski looks like they are disabled by default, ready to rock?

**Comment by djaglowski** [comment]
This looks good to me. Please resolve merge conflicts.

**Comment by sumo-drosiek** [comment]
@djaglowski added to https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/15366

**Comment by djaglowski** [comment]
Closed based on #15366

**Pull request closed without merging**